`TestWatchKeyspace` is flaky

## Flaky Test

### Which jobs are failing
```
--- FAIL: TestClientTestSuite (10.97s)
    --- FAIL: TestClientTestSuite/TestWatchKeyspace (2.81s)
        client_test.go:787: 
            	Error Trace:	client_test.go:787
            	            				client_test.go:732
            	            				keyspace_test.go:72
            	Error:      	Not equal: 
            	            	expected: 0
            	            	actual  : 1
            	Test:       	TestClientTestSuite/TestWatchKeyspace
        client_test.go:742: 
            	Error Trace:	client_test.go:742
            	            				keyspace_test.go:72
            	Error:      	Expected value not to be nil.
            	Test:       	TestClientTestSuite/TestWatchKeyspace
        suite.go:63: test panicked: runtime error: invalid memory address or nil pointer dereference
            goroutine 221279 [running]:
            runtime/debug.Stack()
            	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/debug/stack.go:24 +0x72
            github.com/stretchr/testify/suite.failOnPanic(0xc01122e1a0)
            	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:63 +0x45
            panic({0x381a660, 0x5e095e0})
            	/opt/hostedtoolcache/go/1.19.0/x64/src/runtime/panic.go:890 +0x262
            github.com/tikv/pd/server/cluster.(*RaftCluster).GetStoreConfig(0x0)
            	/home/runner/work/pd/pd/server/cluster/cluster.go:169 +0x3b
            github.com/tikv/pd/tests/client_test.(*clientTestSuite).SetupSuite(0xc0075af290)
            	/home/runner/work/pd/pd/tests/client/client_test.go:754 +0xe57
            github.com/tikv/pd/tests/client_test.(*clientTestSuite).TestWatchKeyspace(0xc0075af290)
            	/home/runner/work/pd/pd/tests/client/keyspace_test.go:72 +0x58
            reflect.Value.call({0xc006e0c9c0?, 0xc00705c408?, 0xe0dc0d?}, {0x3acab5f, 0x4}, {0xc004cc0db8, 0x1, 0xc004cc0be8?})
            	/opt/hostedtoolcache/go/1.19.0/x64/src/reflect/value.go:584 +0xfdd
            reflect.Value.Call({0xc006e0c9c0?, 0xc00705c408?, 0xc0075af290?}, {0xc004cc0db8, 0x1, 0x1})
            	/opt/hostedtoolcache/go/1.19.0/x64/src/reflect/value.go:368 +0xc8
            github.com/stretchr/testify/suite.Run.func1(0xc01122e1a0)
            	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:158 +0x6e7
            testing.tRunner(0xc01122e1a0, 0xc013f44630)
            	/opt/hostedtoolcache/go/1.19.0/x64/src/testing/testing.go:1446 +0x217
            created by testing.(*T).Run
            	/opt/hostedtoolcache/go/1.19.0/x64/src/testing/testing.go:1493 +0x75e
```

### CI link
https://github.com/tikv/pd/runs/8224620941?check_suite_focus=true

### Reason for failure (if possible)


### Anything else

--- a/tests/client/keyspace_test.go
+++ b/tests/client/keyspace_test.go
@@ -67,19 +67,17 @@ func (suite *clientTestSuite) TestLoadKeyspace() {
 }
 
 func (suite *clientTestSuite) TestWatchKeyspace() {
-	// Flush test storage before running TestWatchKeyspace.
-	suite.TearDownSuite()
-	suite.SetupSuite()
 	re := suite.Require()
-	initialKeyspaces := mustMakeTestKeyspaces(re, suite.srv, 0, 10)
+	initialKeyspaces := mustMakeTestKeyspaces(re, suite.srv, 10, 10)
 	watchChan, err := suite.client.WatchKeyspaces(suite.ctx)
 	re.NoError(err)
 	// First batch of watchChan message should contain all existing keyspaces, including the default.
 	initialLoaded := <-watchChan
-	re.Equal(len(initialKeyspaces)+1, len(initialLoaded))
 	for i := range initialKeyspaces {
-		re.Equal(initialKeyspaces[i], initialLoaded[i+1])
+		re.Contains(initialLoaded, initialKeyspaces[i])
 	}
+	keyspaceDefault, err := suite.client.LoadKeyspace(suite.ctx, keyspace.DefaultKeyspaceName)
+	re.Contains(initialLoaded, keyspaceDefault)
 	// Each additional message contains extra put events.
 	additionalKeyspaces := mustMakeTestKeyspaces(re, suite.srv, 30, 10)
 	re.NoError(err)

